pageserver/src/tenant/storage_layer.rs

```diff
@@ -83,6 +89,201 @@ pub enum ValueReconstructResult {
     Missing,
 }
 
+#[derive(Debug)]
+pub struct LayerAccessStats(Mutex<LayerAccessStatsInner>);
+
+#[derive(Debug, Default, Clone)]
+struct LayerAccessStatsInner {
+    first_access: Option<LayerAccessStatFullDetails>,
+    count_by_access_kind: EnumMap<LayerAccessKind, u64>,
+    task_kind_flag: EnumSet<TaskKind>,
+    last_accesses: HistoryBuffer<LayerAccessStatFullDetails, 16>,
+    last_residence_changes: HistoryBuffer<LayerResidenceStatus, 16>,
```

Comment:
For each HistoryBuffer, have a corresponding counter of how many events fell off the end of the buffer.
E.g., when inserting the 17th `last_residence_change`, one falls off at the end. Capture that in a counter, so post-processing at least knows that events are missing.